packages/app-builder-lib/src/util/plist.ts

```diff
@@ -0,0 +1,37 @@
+import { build, parse } from "plist"
+import * as fs from "fs"
+
+function sortObjectKeys(obj: any): any {
+  if (obj === null || typeof obj !== "object") {
+    return obj
+  }
+
+  if (Array.isArray(obj)) {
+    return obj.map(sortObjectKeys)
+  }
+
+  return Object.keys(obj)
+    .sort()
+    .reduce((result: any, key: string) => {
+      result[key] = sortObjectKeys(obj[key])
+      return result
+    }, {})
+}
+
+export async function savePlistFile(path: string, data: any): Promise<void> {
+  const sortedData = sortObjectKeys(data)
+  const plist = build(sortedData)
+  await fs.promises.writeFile(path, plist)
+}
+
+export async function parsePlistFile(file: string): Promise<any> {
+  try {
+    const data = await fs.promises.readFile(file, "utf8")
+    return parse(data)
+  } catch (e: any) {
+    if (e.code === "ENOENT") {
+      return null
```

Comment:
Why this logic? I don't think we should have a `return null`. Also, usages of this function in the code don't seem to handle the `null` return since it's obfuscated by the `Promise<any>`

Reply:
This situation does occur; I've encountered it in the unit tests (UT). Currently, app-builder-bin just returns null(https://github.com/develar/app-builder/blob/580c34f2e19347061bc2243fa6ab6e57e9bbd2a6/pkg/plist/plist.go#L75-L77). If we want to ensure that the file always exists, we would need to check the file path before calling it, which would be more troublesome.
![image](https://github.com/user-attachments/assets/73cadf64-d3db-4ddb-9afe-6ee7516c8b1f)

Reply:
https://github.com/electron-userland/electron-builder/blob/22da64425182456eb4d1243138dde27c80d6adac/packages/app-builder-lib/src/electron/electronMac.ts#L96-L97

Reply:
Fixed. Use safeParsePlistFile.